### DISCUSSION TITLE
i3 & games - will never launch

### DISCUSSION THREAD

**Discussion by NoahRR** [i3/i3#4576] [Q&A]
I absolutely love i3 - but there's one super annoying thing that I can't seem to solve.

I cannot get any games to run. For example: terraria (steam) and league of legends (lutris). They run perfectly when I switch over to GNOME on same setup. (but this is very annoying to do, as I have to close everything I'm doing just to swap over and play a game)

For league, here is the debug log (not very helpful):
lutris-wrapper: League of Legends
Running /home/noahray/.local/share/lutris/runners/wine/lutris-ge-lol-6.16-2-x86_64/bin/wine /home/noahray/Games/league-of-legends/drive_c/Riot Games/League of Legends/LeagueClient.exe --locale=en_US
Initial process has started with pid 434483
Start monitoring process.
wineserver: using server-side synchronization.

Same thing happens with all games: after I click play, they just never load, in a continuous state of loading -- or maybe they are loaded but hidden for some reason?

I've tried installing other dependencies, fiddling around with setting windows to floating / borderless, and tested with other WMs. Maybe this is an issue with tiling window WMs in general. From my testing: works on gnome, does not work on i3 or awesome wm.

Anyone experience similar issues or have any ideas about how to resolve? Any feedback would be greatly appreciated!

(edit) OS: arch (lts kernal)

Let me know if I need to post this question somewhere else - but really trying to find a solution for this!!! I really want to stick with i3!!

**Comment by Airblader** [comment]
What's your compositor situation? Are you running one?

**Reply by NoahRR** [reply]
I'm using picom. Issue was happening before I installed that though.

**Comment by Airblader** [comment]
Does it only affect games running under Wine?

**Reply by NoahRR** [reply]
Not sure - I'm not very versed in the world of linux gaming. But let me try and install some explicitly non-wine games and test it out. Or let me know if you have an easier way to test that. I'll get back to you.

(thanks for the quick responses btw!)

**Reply by NoahRR** [reply]
Hey, I went through some testing and it does seem to be localized to wine / lutris.

CSGO (steam) (native) : works well
TERRARIA (steam) (proton) : works well (started working today)
OVERWATCH (lutris) (wine) : does not open
LEAGUE OF LEGENDS (lutris) (wine) : does not open

I'm finding it difficult to install some games, so sorry for the limited tested games - but hopefully this can be enough to spike some ideas? Let me know if you can think of anything or if there's anything you need me to do as well! Thank you so much!

**Reply by Airblader** [reply]
I'd definitely assume for now that this is related to wine, and given how wine works that makes sense, too. Something about what i3 supports / doesn't support, or how it supports things, could be making wine unhappy.

I don't know if wine also writes its own log files perhaps? Also searching for i3 and wine games might bring up threads of the past when this may have been discussed.

**Reply by NoahRR** [reply]
Right -- ok I'll dig into the issue with a focus on wine & i3. Hopefully that should lead me to an answer! Thanks for the help!

(edit) if I do find a solution that works for me I will post it here for others' future reference

**Reply by Airblader** [reply]
Another thing to try would be running the game under xtrace to see what happens on the X level. This can be somewhat painful to dig through when dealing with complex clients that do lots of things, but if nothing else helps this is something we could try. It'll likely make the client unbearably slow, but since it "crashes" anyway that shouldn't matter.

**Reply by NoahRR** [reply]
right on. I'll try that too!

**Reply by Airblader** [reply]
If you aren't familiar with X you'll probably need our help here to dig through that. If it comes to it you can upload the output somewhere and post the link here.

**Reply by NoahRR** [reply]
You are so helpful! Thanks!

I can't look into this right now, but will do so in the next few days. I'll report back either way (with solution or a linked log).
